components/engine_rocks/src/snapshot.rs

```diff
@@ -30,9 +30,20 @@ impl RocksSnapshot {
             }
         }
     }
+
+    pub fn db(&self) -> Arc<DB> {
+        self.db.clone()
+    }
 }
 
-impl Snapshot for RocksSnapshot {}
+impl Snapshot for RocksSnapshot {
+    type E = RocksEngine;
+
+    #[inline]
+    fn inner_engine(&self) -> Self::E {
```

Comment:
People may assume the returned value is reliable. But actually it's not. All fields except `db` are initialized with the default value.